File: TShock.Plugin.ProjectTemplates/content/Plugin-CSharp/PluginClass.cs
using Terraria;
using TerrariaApi.Server;

namespace Plugin.Namespace;

[ApiVersion(2, 1)]
public class Plugin : TerrariaPlugin
{
    public override string Name => "Plugin.Namespace";
    public override string Author => base.Author;
    public override string Description => base.Description;
    public override Version Version => new Version(1, 0);
    public Plugin(Main game) : base(game)
    {
    }

    public override void Initialize()
    {
        
    }

    protected override void Dispose(bool disposing)
    {
        if (disposing)
        {

        }
    }
}

File: TShock.Plugin.ProjectTemplates/content/Plugin-VisualBasic/PluginClass.vb
Imports Terraria
Imports TerrariaApi.Server

<ApiVersion(2, 1)>
Public Class Plugin 
    Inherits TerrariaPlugin
    Public Overrides ReadOnly Property Name As String = "Plugin.Namespace"
    Public Overrides ReadOnly Property Author As String = "Plugin.Author"
    Public Overrides ReadOnly Property Description As String = "Plugin.Description"
    Public Overrides ReadOnly Property Version As Version = New Version(1, 0)

    Public Sub New(game As Main)
        MyBase.New(game)
    End Sub

    Public Overrides Sub Initialize()

    End Sub

    Protected Overrides Sub Dispose(disposing As Boolean)
        If disposing Then

        End If
    End Sub
End Class
